docs/QUICKSTART.md

```diff
@@ -0,0 +1,410 @@
+    button.mode(PullUp);
+    button.fall(&private_timer_button_on_press);
+
+    /* Setup and start the timer. */
+    Timer timer;
+    timer.start();
+
+    while (1) {
+        /* Store the timer value. */
+        uvisor_ctx->buffer[uvisor_ctx->index] = timer.read_us();
+
+        /* Update the index. Behave as a circular buffer. */
+        if (uvisor_ctx->index < PRIVATE_TIMER_BUFFER_COUNT - 1) {
+            uvisor_ctx->index++;
+        } else {
+            uvisor_ctx->index = 0;
+        }
+    }
+}
+```
+
+A few things to note in the code above:
+
+* If code is running in the context of `private_timer`, then any object instantiated inside that code will belong to the `private_timer` heap and stack. This means that in the example above the `InterruptIn` and `Timer` objects are private to the `private_timer` box. The same applies to the dynamically allocated buffer `uvisor_ctx->buffer`.
+* The content of the private memory `PrivateTimerStaticMemory` can be accessed using the `PrivateTimerStaticMemory * uvisor_ctx` pointer, which is maintained by uVisor.
+* The `InterruptIn` object triggers the registration of an interrupt slot. Since that code is run in the context of the `private_timer` box, then the push-button IRQ belongs to that box. If you want to use the IRQ APIs directly, read the [section](#the-nvic-apis) below.
+* Even if the `private_timer_button_on_press` function runs in the context of `private_timer`, we can still use the `printf` function, which accesses the `UART0` peripheral, owned by the main box. This is because all ACLs declared in the main box are by default shared with all the other secure boxes. This also means that the messages we are printing on the serial port are not secure, because other boxes have access to that peripheral.
+
+> **Note**: Instantiating an object in the `secure_box.cpp` global scope will automatically map it to the main box context (heap), not the `private_timer` one. If you want an object to be private to a box, you need to instantiate it inside the code that will run in the context of that box.
+
+---
+
+**Checkpoint**
+
+Compile the application again, re-flash the device, and press the reset button. The device LED should be blinking as in the previous case.
+
+If you don't see the LED blinking, it means that the application halted somewhere, probably because uVisor captured a fault. You can setup the uVisor debug messages to see if there is any problem. Follow the [Debugging uVisor](DEBUGGING.md) document for a step-by-step guide.
+
+If the LED is blinking, it means that the app is running fine. If you now press the `SW2` button on the NXP FRDM-K64F board, the `private_timer_button_on_press` function will be executed, printing the values in the timer buffer. You can observe these values by opening a serial port connection to the device, with a baud rate of 9600. When the print is completed, you should see the LED blinking again.
+
+### Allow other boxes to talk to the secure box
```

Comment:
Boxes can insecurely talk to each other through the _main box_. Perhaps rename this section to `Allow other boxes to securely communicate to the secure box`.

Why `secure box` and not `secure boxes`? If because there is only one secure box in this example, then consider saying `main box` instead of `other boxes` as that's the only other box.